### PR TITLE
[improve][doc] Add doc for builtin functions

### PR DESCRIPTION
**Pull request by cbornet**
### Documentation

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/cbornet/pulsar/pull/9

### CODE REVIEW

File: site2/docs/functions-deploy-cluster-builtin.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+bin/pulsar-admin functions available-functions
+```
+
+If you add or delete a nar file in a `functions` folder, reload the available built-in functions before using it.
+
+```bash
+bin/pulsar-admin functions reload
+```
+
+
```
Comment (tisonkun) [inline site2/docs/functions-deploy-cluster-builtin.md:30]:
```suggestion
```

Trim whitespace.

File: site2/docs/functions-package-java.md [hunk 2]
```diff
@@ -10,7 +10,23 @@ For the runtime Java version, refer to [Pulsar Runtime Java Version Recommendati
...
+
+If you plan to package and distribute your function for others to use, you are obligated to
+license and copyright your own code properly. Remember to add the license and copyright to
+all libraries your code uses and to your distribution.
+
+:::
+
+> If you use the [NAR](#package-as-nar) method, the NAR plugin
+automatically creates a `DEPENDENCIES` file in the generated NAR package, including the proper
+licensing and copyrights of all libraries of your function.
```
Comment (tisonkun) [inline site2/docs/functions-package-java.md:25]: They seem similar notes/tips, why different syntax?

You may read the syntax guide (https://pulsar.apache.org/contribute/document-syntax/#admonitions) for what is expected here.

File: site2/docs/functions-package-java.md [hunk 3]
```diff
@@ -106,3 +122,90 @@ To package a Java function, complete the following steps.
...
+
+        <groupId>java-function</groupId>
+        <artifactId>java-function</artifactId>
+        <version>1.0-SNAPSHOT</version>
+
+        <dependencies>
+            <dependency>
+                <groupId>org.apache.pulsar</groupId>
+                <artifactId>pulsar-functions-api</artifactId>
+                <version>2.10.0</version>
```
Comment (tisonkun) [inline site2/docs/functions-package-java.md:147]:
```suggestion
                <version>@pulsar:version@</version>
```

Perhaps resolve to the lastest version.

File: site2/docs/functions-package-java.md [hunk 4]
```diff
@@ -106,3 +122,90 @@ To package a Java function, complete the following steps.
...
+    </project>
+   ```
+
+You must also create a `resources/META-INF/services/pulsar-io.yaml` file. In the following code sample, the value of `functionClass` is your function class name. The `name` is the one used when the Function is deployed as a [built-in](functions-deploy-cluster-builtin.md) one.
+
+```yaml
+name: java-function
+description: my java function
+functionClass: org.example.test.ExclamationFunction
+```
```
Comment (tisonkun) [inline site2/docs/functions-package-java.md:177]: Also one-level ident?

File: site2/docs/functions-package-java.md [hunk 5]
```diff
@@ -106,3 +122,90 @@ To package a Java function, complete the following steps.
...
+
+2. Package your Java function.
+
+   ```bash
+    mvn package
```
Comment (tisonkun) [inline site2/docs/functions-package-java.md:182]:
```suggestion
   mvn package
```

Remove leading whitespace. Ditto others.

**Review by tisonkun — COMMENTED** [review]
Thanks for your contribution @cbornet! Comments inline.

Also, you need to add the `functions-deploy-cluster-builtin` doc to [sidebars.json](https://github.com/apache/pulsar/blob/master/site2/website/sidebars.json) so that it shows on the sidebar.

BTW, you can read [the Preview Guide](https://pulsar.apache.org/contribute/document-preview) to preview locally.

File: site2/docs/functions-package-java.md (hunk 4 above)
Comment (cbornet) [reply site2/docs/functions-package-java.md]: done

File: site2/docs/functions-package-java.md (hunk 2 above)
Comment (cbornet) [reply site2/docs/functions-package-java.md]: done

**Review by tisonkun — APPROVED** [review]
LGTM.

**Comment by cbornet** [comment]
I removed usage of `available-functions` and `reload` as they are not part of next/2.11

File: site2/docs/functions-package-java.md [hunk 6]
```diff
@@ -4,13 +4,33 @@ title: Package Java Functions
 sidebar_label: "Package Java Functions"
 ---
 
+There are two methods to package Java Functions, that is [uber JAR](#package-as-jar) and [NAR](#package-as-nar).
+
+:::note
```
Comment (momo-jun) [inline site2/docs/functions-package-java.md:9]: With the note and tip newly added, there will be three consecutive color blocks (but in different colors), which is a bit overwhelming in visuals IMO. Does it help by combining the three items in a single note/tip block?

File: site2/docs/functions-deploy-cluster-builtin.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
+---
+id: functions-deploy-cluster-builtin
```
Comment (momo-jun) [inline site2/docs/functions-deploy-cluster-builtin.md:2]: Suggest adding the link of this new topic to the [parent topic](https://pulsar.apache.org/docs/next/functions-deploy-cluster) to update the map for users.

File: site2/docs/functions-deploy-cluster-builtin.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+title: Built-in functions
+sidebar_label: "Built-in functions"
```
Comment (momo-jun) [inline site2/docs/functions-deploy-cluster-builtin.md:4]: It's a task-oriented topic for users. So it makes more sense to use an imperative title, like `Use built-in functions`.

File: site2/docs/functions-package-java.md (hunk 6 above)
Comment (cbornet) [reply site2/docs/functions-package-java.md:9]: done

File: site2/docs/functions-deploy-cluster-builtin.md (hunk 7 above)
Comment (cbornet) [reply site2/docs/functions-deploy-cluster-builtin.md:2]: done

File: site2/docs/functions-deploy-cluster-builtin.md (hunk 8 above)
Comment (cbornet) [reply site2/docs/functions-deploy-cluster-builtin.md]: done

**Review by momo-jun — APPROVED** [review]
LGTM. 
Thanks for adding the docs. It deserves a `feature` rather than `Improve` in the PR title :)

**Comment by tisonkun** [comment]
Meriging...

**Pull request merged**